src/parser/parsed_data/drop_secret_info.cpp

```diff
@@ -0,0 +1,17 @@
+#include "duckdb/parser/parsed_data/drop_secret_info.hpp"
+
+namespace duckdb {
+
+DropSecretInfo::DropSecretInfo() : DropInfo(), persist_mode(SecretPersistType::DEFAULT), secret_storage("") {
```

Comment:
Instead of using inheritance of `DropInfo` here perhaps we can have an `ExtraDropInfo` member variable that `DropInfo` has and that can be subclassed by the `DropSecretInfo` - this way we can avoid the special case serialization code.